rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,356 @@
+server will use the STS [assume role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)
+functionality to generate a token with temporary access to upload to S3. This token is referred to as an access token.
+
+The access token generated and provided grants direct access to the S3 bucket for uploading. This will allow
+the browser APP to use the AWS JS SDK to perform the multipart uploads to S3 for large files. A backend sever will be used
+to generate the token and additional sessions policies will be applied to the token to further restricts access
+
+According to AWS:
+
+> The resulting session's permissions are the intersection of the role's identity-based policies and the session policies.
+
+The IAM policies bellow will be combined to limit the access of the token generate to only allow the bearer to upload the
+file in the session policy resource. The session policy will be generated at the time of the request, and will use the
+asset_uuid generated by the server and the data set_uuid provided in the request.
+
+##### Assumed Role IAM Policy
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListMultipartUploadParts",
+        "s3:PutObject",
+        "s3:AbortMultipartUpload"
+      ],
+      "Resource": ["arn:aws:s3:::upload-bucket/"]
+    }
+  ]
+}
+```
+
+##### IAM Session Policy
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListMultipartUploadParts",
+        "s3:PutObject",
+        "s3:AbortMultipartUpload"
+      ],
+      "Resource": ["arn:aws:s3:::upload-bucket/{data set_uuid}/{asset_uuid}"]
+    }
+  ]
+}
+```
+
+#### EC2
+
+The EC2 instance will be used to download files from a CSP and upload to our Upload bucket.
+
+#### Upload Bucket
+
+The upload bucket is a separate bucket or separate bucket location within our main bucket. It is used to stage files
+before moving to the main bucket.
+
+S3 event can be use to trigger additional processing such as validation, once an upload has completed. The additional
+processing is outside the scope of this RFC.
+
+### Data Portal APIs
+
+The DP Browser App will use these endpoint to interact with the Backend using these APIs. This section describes the new
+endpoints and how they are used. These are the new API endpoints needed to complete an upload.
+
+These APIs replace the [PUT /v1/submission/{project_uuid}/file](https://docs.google.com/document/d/1d8tv2Ub5b3E7Il85adOAUcG8P05N6UBZJ3XbhJSRrFs/edit#heading=h.3hln6w2kzoyt)
+section.
+
+#### PUT collection/{collection_id}/{dataset_id}/upload/link
+
+An authenticated user can upload a file from a shared link to a data set in their collection.
+
+##### Request: POST collection/{collection_id}/{dataset_id}/upload/link
+
+- Link - a shared link to the file
+- collection_id - identifies the collection
+- dataset_id - identifies the data set
```

Comment:
Is this required? For a new dataset being uploaded, this can be created by the backend right?

Reply:
you are right. I fixed the APIs